src/loaders/genome_collection/compute_genome_attribs.py

```diff
@@ -303,9 +342,36 @@ def checkm2(genome_ids, work_dir, source_data_dir, debug, program_threads, batch
     _run_command(command, debug=debug, log_dir=os.path.join(batch_dir, 'checkm2_log'))
     end_time = time.time()
     print(
-        f'Used {round((end_time - start) / 60, 2)} minutes to execute checkM2 predict for {len(genome_ids)} genomes')
+        f'Used {round((end_time - start) / 60, 2)} minutes to execute checkM2 predict for {size} genomes')
+
+    _create_genome_metadata_file(genomes_meta, batch_dir)
+    # TODO: inspect stdout for failed ids or do it in the parser program
+    return failed_ids
+
+
+def microtrait(genome_ids, work_dir, source_data_dir, debug, program_threads, node_id, source_file_ext,
+               kbase_collection):
+    failed_ids, size = list(), len(genome_ids)
+    print(f'Start executing MicroTrait for {size} genomes')
+
+    batch_dir, genomes_meta = _prepare_tool('microtrait', work_dir, 'series', size, node_id, genome_ids,
+                                            source_data_dir, source_file_ext, kbase_collection)
+    start = time.time()
+
+    # RUN MicroTrait in parallel with multiprocessing
+    args_list = [(genome_id, genome_meta['source_file'], debug) for genome_id, genome_meta in genomes_meta.items()]
+    pool = multiprocessing.Pool(processes=program_threads)
+    results = pool.starmap(_run_microtrait, args_list)
+    pool.close()
+    pool.join()
+
+    # TODO process results from _run_microtrait
+
+    end_time = time.time()
+    print(
+        f'Used {round((end_time - start) / 60, 2)} minutes to execute MicroTrait for {size} genomes')
 
-    _create_genome_metadata_file(tool_genome_id_map, source_genome_file_map, len(genome_ids), batch_dir)
+    _create_genome_metadata_file(genomes_meta, batch_dir)
     # TODO: inspect stdout for failed ids or do it in the parser program
```

Comment:
To my mind this should be done here so the job is registered as a fail and can be restarted. Is this something that'll show up in a a PR or two? If not I'm thinking we should make a ticket, I don't want to lose track of this

Reply:
we do have multiple tickets in the backlog regarding rerun incomplete genomes. I remember during one of the meetings, we agreed that it's not a pressing concern at the moment, as it would actually be an improvement since we can simply restart the entire task.
https://kbase-jira.atlassian.net/browse/RE2022-133
https://kbase-jira.atlassian.net/browse/RE2022-96

Reply:
Right, but this is a little different - it's causing the job to fail so that it can be rerun. Those tickets are about rerunning failed jobs, this TODO is about making the job fail when it should

Reply:
What is this TODO for then?

Reply:
just to remind me that I need to fix those failed ids.

Reply:
Per phone convo, these TODOs are no longer relevant, will be removed

Reply:
👍

Reply:
:+1:

Reply:
To explain for posterity, the failed_ids here referred to genome_ids without fasta files (e.g. empty directories). If a file fails to process otherwise, the job will fail per @Tianhao-Gu